### PR TITLE
[RHPAM-4781] KIE-OSGI: Inconsistent h2 version upgrade

### PR DESCRIPTION
**Pull request by tkobayas**
**JIRA**: 
https://issues.redhat.com/browse/RHPAM-4781

This PR is for main(= 7.x)
for 7.67.x ->
for 7.67.x-blue ->

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

File: kie-osgi/kie-karaf-features/pom.xml [hunk 1]
```diff
@@ -47,9 +47,11 @@
     <karaf.version.org.apache.servicemix.bundles.ant>1.8.2_2</karaf.version.org.apache.servicemix.bundles.ant>
     <karaf.version.org.apache.servicemix.bundles.dom4j>1.6.1_5</karaf.version.org.apache.servicemix.bundles.dom4j>
     <karaf.version.org.apache.servicemix.bundles.serp>1.14.1_1</karaf.version.org.apache.servicemix.bundles.serp>
-    <karaf.version.com.fasterxml.classmate>1.3.1</karaf.version.com.fasterxml.classmate>
+    <karaf.version.com.fasterxml.classmate>1.5.1</karaf.version.com.fasterxml.classmate>
     <karaf.version.org.jboss.jandex>2.2.3.Final</karaf.version.org.jboss.jandex>
     <karaf.version.javax.validation>1.1.0.Final</karaf.version.javax.validation>
+    <karaf.version.org.javassist>3.27.0-GA</karaf.version.org.javassist>
+    <karaf.version.org.hibernate.commons.annotations>5.1.2.Final</karaf.version.org.hibernate.commons.annotations>
```
Comment (tkobayas) [inline kie-osgi/kie-karaf-features/pom.xml:54]: Apart from H2 version, I had to specify these versions to meet the hibernate-core version `5.4.24.Final`'s dependencies.

**Comment by tkobayas** [comment]
GHA test failures are not related to this PR.

Java 8
```
2023-10-23T09:42:04.0332698Z [ERROR] Tests run: 96, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 267.829 s <<< FAILURE! - in org.kie.server.integrationtests.jbpm.JobServiceIntegrationTest
```

Java 11
```
2023-10-23T08:44:55.6413955Z [ERROR] Tests run: 15, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 13.872 s <<< FAILURE! - in org.kie.server.client.LoadBalancerClientTest
```

I confirmed that kie-karaf-itests were successful.

Java 8
```
2023-10-23T10:47:24.7929943Z [INFO] --- maven-failsafe-plugin:2.22.1:integration-test (default) @ kie-karaf-itests ---
2023-10-23T10:47:24.8009385Z [INFO] Failsafe report directory: /home/runner/work/droolsjbpm-integration/droolsjbpm-integration/kiegroup_droolsjbpm-integration/kie-osgi/kie-karaf-itests/target/failsafe-reports
2023-10-23T10:47:24.8034508Z [INFO]
2023-10-23T10:47:24.8056209Z [INFO] -------------------------------------------------------
2023-10-23T10:47:24.8064033Z [INFO]  T E S T S
2023-10-23T10:47:24.8084017Z [INFO] -------------------------------------------------------
2023-10-23T10:47:25.2484370Z [INFO] Running org.kie.karaf.itest.InstallFeaturesKarafIntegrationTest
2023-10-23T10:48:12.4055759Z [INFO] Tests run: 20, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 47.153 s - in org.kie.karaf.itest.InstallFeaturesKarafIntegrationTest
...
2023-10-23T11:03:07.5117459Z [INFO] Results:
2023-10-23T11:03:07.5117852Z [INFO]
2023-10-23T11:03:07.5118354Z [WARNING] Tests run: 62, Failures: 0, Errors: 0, Skipped: 2
```

**Comment by tkobayas** [comment]
@grgrzybek @skulisek Please review. I tested only with upstream Karaf. For Fuse, I will test in a backport PR for 7.67.x branch.

**Review by grgrzybek — APPROVED** [review]
Checked with Karaf and Fuse-Karaf. all tests pass.

**Comment by skulisek** [comment]
Looks fine to me

**Review by skulisek — APPROVED** [review]

**Comment by tkobayas** [comment]
@mariofusco Could you merge this PR? Thanks!

**Review by mariofusco — APPROVED** [review]

**Pull request merged**